File: Data Structures & Algorithms/binary-search/submission-4.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let minRange = 0;
        let maxRange = nums.length - 1;
        let i = Math.floor((minRange + maxRange) / 2);

        while (minRange <= maxRange) {
            console.log(i, nums[i], target);
            if (nums[i] > target) {
                maxRange = i - 1
                i = Math.floor((minRange + maxRange) / 2);
            }

            if (nums[i] < target) {
                minRange = i + 1
                i = Math.floor((minRange + maxRange) / 2);

            }

            if (nums[i] === target) {
                return i
            }
        }

        return -1
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-4.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0;
        int r = numbers.length - 1;

        while(l<r) {
            int currentTotal = numbers[l] + numbers[r];

            if(currentTotal < target) {
                l++;
            } else if (currentTotal > target) {
                r--;
            } else {
                return new int[] {l+1, r+1};
            }
        }

        return new int[] {};
    }
}
